modules/gke-hub/main.tf

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+locals {
+  config_sync_sa_email = (
+    local.hub_config.config_sync.workload_identity_sa == null
+    ? (
+      length(google_service_account.gke-config-management-wid-sa[0]) > 0
+      ? google_service_account.gke-config-management-wid-sa[0].email
+      : null
+    )
+    : local.hub_config.config_sync.workload_identity_sa
+  )
+  config_sync_repository_url = (
+    local.hub_config.config_sync.repository_url == null
+    ? (
+      length(google_sourcerepo_repository.default[0]) > 0
+      ? google_sourcerepo_repository.default[0].url
+      : null
+    )
+    : local.hub_config.config_sync.repository_url
+  )
+
+  config_sync       = coalesce(var.hub_config.config_sync, var.config_sync_defaults)
+  policy_controller = coalesce(var.hub_config.policy_controller, var.policy_controller_defaults)
+  hub_config = {
+    clusters          = var.hub_config.clusters
+    config_sync       = merge(var.hub_config.config_sync, local.config_sync)
+    policy_controller = merge(var.hub_config.policy_controller, local.policy_controller)
+  }
+}
+
+#############################################################################
+# Add each clusters to the GKE Hub by creating the membership/s
+#############################################################################
+resource "google_gke_hub_membership" "membership" {
+  provider      = google-beta
+  for_each      = { for i, v in local.hub_config.clusters : i => v }
+  membership_id = each.value.name
+  project       = var.project_id
+  endpoint {
+    gke_cluster {
+      resource_link = "//container.googleapis.com/projects/${var.project_id}/locations/${each.value.location}/clusters/${each.value.name}"
+    }
+  }
+}
+
+#############################################################################
+# Enable configmanagement feature on the cluster/s registered into the hub
+#############################################################################
+resource "google_gke_hub_feature" "feature" {
```

Comment:
what about other features? There's at least two more, but I can't find a canonical list.

Reply:
Added all the remaining features